### DISCUSSION TITLE
[Question] How to create a staging instance with doodba?

### DISCUSSION THREAD

**Discussion by obayit** [Tecnativa/doodba#356] [Q&A] [answered]
Not sure if this is a question, or a feature request.

A staging is a copy of the production database with a PR applied to it from a specific repository.

The "brute force" approach would be to copy the project directory into a new one. Then change the `.copier-answers.yml` values like `domains_prod`, `domains_test`, `postgres_dbname?`. Then add the PR in `repos.yaml`. And finally build and run with `prod.yaml`.

Another approach may be to use a different `.copier-answers.yml` file, and change `repos.yaml` temporarily to add the PR and run a new instance of the `prod.yaml`.

My question is, how to create a staging instance with doodba?

**Comment by pedrobaeza** [comment, marked as answer]
Copy the scaffolding directory, and inside, change docker-compose.yml link from prod.yaml to test.yaml. Then you just need to restore the DB/filestore over it for having a staging system isolated from the outside.

**Reply by rquevedo** [reply]
> Copy the scaffolding directory, and inside, change docker-compose.yml link from prod.yaml to test.yaml. Then you just need to restore the DB/filestore over it for having a staging system isolated from the outside.

How can i restore a copy of DB/filestore if backup container dont start in test mode. By other hand i have db_list=False in prod by security reasons.

**Reply by obayit** [reply]
To restore the database you have to do it manually.
```
docker-compose -f test.yaml run --rm -l traefik.enable=false odoo psql -q < DB_DUMP_FILE_PATH > /dev/null 2>&1
```

**Reply by rquevedo** [reply]
Hi @obayit i dont have DB_DUMP of production environment. I only have backups made with duplicity.

**Comment by obayit** [comment]
Thanks, I didn't even know that `docker-compose.yml` was a link :laughing:
